### PR TITLE
Added PolyLine Features and updated utils with extra features

### PR DESCRIPTION
**Pull request by kengoon**
This pull request contains a PolyLineLayer that can be added to MapView and has the ability to adjust the mapview to a certain geo-coordinates with a reasonable zoom level, in case of coordinates update or change

### CODE REVIEW

File: kivy_garden/mapview/utils.py [hunk 1]
```diff
@@ -43,8 +44,191 @@ def get_zoom_for_radius(radius_km, lat=None, tile_size=256.0):
...
+        `map_width`: Width of the map
+        `map_height`: Height of the map
+
+    :return: Calculated zoom level.
+    """
+    min_lat, max_lat, min_lon, max_lon = get_bounding_box(locations)
+
+    # map_width /= 5
+    # map_height /= 5
+
```
Comment (tito) [inline kivy_garden/mapview/utils.py]: To be removed

File: kivy_garden/mapview/utils.py [hunk 2]
```diff
@@ -43,8 +44,191 @@ def get_zoom_for_radius(radius_km, lat=None, tile_size=256.0):
...
+    # Constants for zoom calculation
+    TILE_SIZE = 256
```
Comment (tito) [inline kivy_garden/mapview/utils.py]: Please pass it as a parameter, with a default of 256 (use `tile_size=256`), as the library can be used with other source size (512 for retina screen for eg)

File: kivy_garden/mapview/utils.py [hunk 3]
```diff
@@ -43,8 +44,191 @@ def get_zoom_for_radius(radius_km, lat=None, tile_size=256.0):
...
+        map_width,
+        map_height,
+        lat1,
+        lon1,
+        lat2,
+        lon2,
+        mapview=None,
+        polyline_layer=None,
+        max_zoom=16
+
```
Comment (tito) [inline kivy_garden/mapview/utils.py]: Same here, please pass the tile_size as well, and pass it to the function that accept it

File: kivy_garden/mapview/utils.py [hunk 4]
```diff
@@ -43,8 +44,191 @@ def get_zoom_for_radius(radius_km, lat=None, tile_size=256.0):
...
+    zoom_level = int((z1 + z2) / 2)
+    if mapview:
+        mapview.center_on(center_lat, center_lon)
+        mapview.zoom = min(zoom_level, max_zoom)
+    if polyline_layer:
+        polyline_layer.coordinates = coordinates
+    return (center_lat, center_lon), zoom_level
+
+
+def findpoints(lat, lon, radius):
```
Comment (tito) [inline kivy_garden/mapview/utils.py]: Please rename the function to something that match the intent. Something like `generate_circle_points()`.
Add the precision/count (currently N) as part of the parameters.

File: kivy_garden/mapview/view.py [hunk 5]
```diff
@@ -273,6 +273,24 @@ def unload(self):
...
+        mapview = self.parent
+        with self.canvas:
+            self.canvas.clear()
+            Color(1, 0, 0, 1)  # Red color for the polyline
+            points = []
+            for lat, lon in self.coordinates:
+                x, y = mapview.get_window_xy_from(lat, lon, mapview.zoom)
+                points.extend([x, y])
+            if points:
+                SmoothLine(points=points, width=2, joint='round', cap='round')
```
Comment (tito) [inline kivy_garden/mapview/view.py]: It would be better to cache the SmoothLine and just update the `.points` attribute on `reposition()`.

**Review by tito — CHANGES_REQUESTED** [review]

**Comment by kengoon** [comment]
@tito all clean now. Go ahead and check it out

**Pull request merged**